package/MDAnalysis/analysis/polymer.py

```diff
@@ -0,0 +1,220 @@
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        skip : int, optional
+            Step between frames to analyse, Default: 1
+        """
+        self._atomgroups = atomgroups
+
+        # Check that all chains are the same length
+        lens = [len(ag) for ag in atomgroups]
+        chainlength = len(atomgroups[0])
+        if not all([l == chainlength for l in lens]):
+            raise ValueError("Not all AtomGroups were the same size")
+
+        kwargs.update({'traj': atomgroups[0].universe.trajectory})
+        self._setup_frames(**kwargs)
+
+        self._results = np.zeros(chainlength - 1, dtype=np.float32)
+
+    def _single_frame(self):
+        # could optimise this by writing a "self dot array"
+        # we're only using the upper triangle of np.inner
+        # function would accept a bunch of coordinates and spit out the
+        # decorrel for that 
+        n = len(self._atomgroups[0])
+
+        for chain in self._atomgroups:
+            # Vector from each atom to next
+            vecs = chain.positions[1:] - chain.positions[:-1]
+            # Normalised to unit vectors
+            vecs /= np.sqrt((vecs * vecs).sum(axis=1))[:, None]
+
+            inner_pr = np.inner(vecs, vecs)
+            for i in xrange(n-1):
+                self._results[:(n-1)-i] += inner_pr[i, i:]
+
+    def _normalise(self):
+        n = len(self._atomgroups[0])
+
+        norm = np.linspace(n - 1, 1, n - 1)
+        norm *= len(self._atomgroups) * len(self.frames)
+
+        self.results = self._results / norm
+        self._calc_bond_length()
+
+    def _calc_bond_length(self):
+        """calculate average bond length"""
+        bs = []
+        for ag in self._atomgroups:
+            pos = ag.positions
+            b = calc_bonds(pos[:-1], pos[1:]).mean()
+            bs.append(b)
+        self.lb = np.mean(bs)
+
+    def fit(self):
+        """Fit the results to an exponential decay"""
+        from scipy.optimize import curve_fit
+
+        try:
+            results = self.results
+        except AttributeError:
+            raise ValueError("Use the run method first")
+        self.x = np.arange(len(self.results)) * self.lb
+
+        self.lp = fit_exponential_decay(self.x, self.results)
+
+        self.fit = np.exp(-self.x/self.lp)
+
+    def plot(self):
+        """Oooh fancy"""
+        import matplotlib.pyplot as plt
+
+        plt.plot(self.x, self.results, 'ro')
+        plt.plot(self.x, self.fit)
+        plt.show()
+
+
+def fit_exponential_decay(x, y):
+    r"""Fit a function to an exponential decay
+
+    .. math::  y = \exp(-x/a)
+
+    Parameters
+    ----------
+    x, y : array_like
+      The two arrays of data
+
+    Returns
+    -------
+    a : float
+      The coefficient *a* for this decay
+
+    Notes
+    -----
+    Requires scipy
+    """
+    from scipy.optimize import curve_fit
+
+    def expfunc(x, a):
```

Comment:
you should also fit the normalization constant of the exponential. You can't expect that everyone using this will always have done the normalization before, in some cases this is also not possible.